cmd/xl-sets.go

```diff
@@ -1010,6 +1010,99 @@ func (s *xlSets) ReloadFormat(ctx context.Context, dryRun bool) (err error) {
 	return nil
 }
 
+// FixDeploymentID - Add deployment id if it is not present.
+func (s *xlSets) FixDeploymentID(ctx context.Context) (err error) {
+	// Acquire lock on format.json
+	formatLock := s.getHashedSet(formatConfigFile).nsMutex.NewNSLock(minioMetaBucket, formatConfigFile)
+	if err = formatLock.GetLock(globalHealingTimeout); err != nil {
+		return err
+	}
+	defer formatLock.Unlock()
+
+	storageDisks, err := initStorageDisks(s.endpoints)
+	if err != nil {
+		return err
+	}
+
+	defer func(storageDisks []StorageAPI) {
+		if err != nil {
+			closeStorageDisks(storageDisks)
```

Comment:
It looks like the purpose of closeStorageDisks() is just to stop go routines which are responsible of updating storage metrics.. I guess we need to call closeStorageDisks() in all cases and not only when `FixDeploymentID` returns an error.

Reply:
Just to be clear, you want to call `closeStorageDisks` in all cases only when an error is to be returned. In other cases, the existing code is fine?

Reply:
No FixDeploymentID rewrites the storageDisks so it shouldn't close them upon success @vadmeste @kannappanr